src/backend/columnar/columnar_storage.c

```diff
@@ -347,10 +347,32 @@ ColumnarStorageIsCurrent(Relation rel)
 
 
 /*
- * ColumnarStorageReserveStripe - reserve stripe ID and row numbers.
+ * ColumnarStorageReserveRowNumber returns reservedRowNumber and advances
+ * it for next row number reservation.
  */
 uint64
-ColumnarStorageReserveStripe(Relation rel, uint64 nrows, uint64 *firstRowNumber)
+ColumnarStorageReserveRowNumber(Relation rel, uint64 nrows)
```

Comment:
Did you separate these for readability reasons, or is there a reason that they need to happen at different times?

Reply:
Second one. As we discussed before, we can't defer reserving `reservedRowNumber` to stripe reservation time since we need to set tid for tuples during insert.
That way, indexAM build callback would work fine after insert/copy etc.

Reply:
The other option would be to reserve the stripe ID earlier at insert time. That might be ugly. I'm not objecting to them being separate but I wanted to clarify.